python_modules/dagster/dagster/_core/definitions/unresolved_asset_job_definition.py

```diff
@@ -121,11 +123,22 @@ def run_request_for_partition(
         Returns:
             RunRequest: an object that requests a run to process the given partition.
         """
+        from dagster._core.definitions.mutable_partitions_definition import (
+            MutablePartitionsDefinition,
+        )
+
         partition_set = self.get_partition_set_def()
         if not partition_set:
             check.failed("Called run_request_for_partition on a non-partitioned job")
 
-        partition = partition_set.get_partition(partition_key)
+        if isinstance(partition_set.partitions_def, MutablePartitionsDefinition):
+            if not instance:
```

Comment:
I think if we removed this code it would throw an error from `MutablePartitionsDefinition.get_partitions`. My hesitation was just that it'll throw a `check.failed(""Must provide a dagster instance object or mutable partitions store to fetch mutable partitions")` and it'll be unclear that the instance needs to be provided through `run_request_for_partitions`.

The hope down the line is that we can validate partitions when processing run requests so the user doesn't need to provide an instance through `run_request_for_partitions`, but I was hoping to break that out into another PR.

Reply:
Gotcha - OK, agreed that's a better error message